narwhals/_arrow/utils.py

```diff
@@ -76,6 +77,17 @@ def native_to_narwhals_dtype(dtype: Any, dtypes: DTypes) -> DType:
 
 
 def narwhals_to_native_dtype(dtype: DType | type[DType], dtypes: DTypes) -> Any:
+    if (pl := get_polars()) is not None and isinstance(
+        dtype, (pl.DataType, pl.DataType.__class__)
```

Comment:
`pl.DataType.__class__` is a workaround since `pl.DataTypeClass` now raises a DeprecationWarning.

I think this is the best way to check for both `pl.String()` and `pl.String`: `if "polars" in str(type(dtype))` was definitly not enough for both

Reply:
nice, thanks!